syft/internal/cache/dir.go

```diff
@@ -0,0 +1,71 @@
+package cache
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/adrg/xdg"
+	"github.com/mitchellh/go-homedir"
+
+	"github.com/anchore/syft/internal/log"
+)
+
+// SetRootDir sets the root directory to use for downloaded content caching for things like module license information
+func SetRootDir(dir string) {
+	rootDir = dir
+}
+
+// rootDir if set, will be the default root directory used to cache content
+var rootDir = ""
+
+// findRoot returns a directory to be used for caching, first available in this order:
+//   - rootDir
+//   - XDG_CACHE_HOME
+//   - HOME/.cache
+//   - TEMP
+func findRoot() string {
+	if rootDir != "" {
+		dir, err := homedir.Expand(rootDir)
+		if err != nil {
+			log.Warnf("unable to expand cache directory %s: %v", rootDir, err)
+		} else {
+			return dir
+		}
+	}
+
+	var err error
+	cacheRoot := xdg.CacheHome
+	if cacheRoot == "" {
+		cacheRoot, err = homedir.Dir()
+		if err != nil {
+			cacheRoot = os.TempDir()
+			log.Debugf("unable to get home directory due to: %v, defaulting cache to temp dir: %s", err, cacheRoot)
+		} else {
+			cacheRoot = filepath.Join(cacheRoot, ".cache")
+		}
+	}
+
+	return filepath.Join(cacheRoot, "syft")
+}
+
+// getDir returns a writable directory with the given name under the root cache directory returned from findRoot,
+// the directory will be created if it does not exist
+func getDir(subDirs ...string) (string, error) {
+	dir := filepath.Join(findRoot(), filepath.Join(subDirs...))
+	stat, err := os.Stat(dir)
+	if errors.Is(err, os.ErrNotExist) {
+		err = os.MkdirAll(dir, os.ModeDir|0700)
+		stat, _ = os.Stat(dir)
```

Comment:
I think this error should also be checked, since the value of `stat` is depended on later (and shouldn't be if this errored out)